framework/src/utils/SlepcSupport.C

```diff
@@ -106,6 +110,9 @@ getSlepcEigenProblemValidParams()
 
   params.addParam<unsigned int>("free_power_iterations", 4, "The number of free power iterations");
 
+  params.addParam<unsigned int>(
```

Comment:
We need a protection here: out PJFNK eigen solver only can find the smallest eigenvalue, so we should error out if users set the `which_eigen_pairs` to something like `LARGEST_MAGNITUDE`. In more general sense, we need to check if a specific eigen solver in SLEPc can deliver the desired eigen pairs.

Reply:
In general, we should not protect this in MOOSE, instead, it should be handled by SLEPc/PETSc.  For example, right now, PJFNK only makes sense for SMALLEST_MAGNITUDE (that was chosen by default). SLEPc will be angry we choose wrong parameters. In addition, in the future, PJFNK can be enhanced  to find other spectrums.

Reply:
Is SLEPc erroring out on the wrong settings then?

Reply:
I think so, otherwise thing is really wrong